090-add-connect-to-test-container.md

```diff
@@ -0,0 +1,153 @@
+# Support running Kafka Connect in test-container
+
+The Strimzi [test-container](https://github.com/strimzi/test-container) library allows running Kafka clusters in containers. This is useful for integration and system tests that require a cluster.
+
+This proposes adding support for running Kafka Connect clusters. 
+
+## Motivation
+
+With [test-container](https://github.com/strimzi/test-container) and [test-clients](https://github.com/strimzi/test-clients), it's possible to build test environments for all Apache Kafka components but Kafka Connect. 
+
+Being able to easily start Kafka Connect clusters would be useful for testing client-side components, such as [metrics-reporter](https://github.com/strimzi/metrics-reporter) and [strimzi-kafka-oauth](https://github.com/strimzi/strimzi-kafka-oauth), with Kafka Connect.
+
+## Proposal
+
+Create 2 new classes in the `io.strimzi.test.container` package of `test-container`:
+- `StrimziConnectCluster`: This represents a Kafka Connect cluster. This is the main API users will call to create and interact with Kafka Connect clusters.
+- `StrimziConnectContainer`: This represents a Kafka Connect worker.
```

Comment:
Do we really need two different public classes? Wouldn't `StrimziConnectCluster` be sufficient? For the Kafka clusters, this duality seems weird and tends to each class supporting different features etc.

Reply:
Yeah, I think maybe if we just expose `StrimziConnectCluster` and having core functionality in the `StrimziConnectContainer` class as we have it for `StrimziKafkaContainer` could be beneficial ... in terms that you can easily create StrimziConnectCluster with 1 replica (i.e., `StrimziConnectContainer`) or multiple replicas...

Reply:
What is hard on using the `StrimziConnectCluster` cluster with single replica? I think we should not have two public APIs to do the same as it proved very disfunctional for the Kafka brokers.

Reply:
> What is hard on using the StrimziConnectCluster cluster with single replica? I think we should not have two public APIs to do the same as it proved very disfunctional for the Kafka brokers.

By the previous comment, I meant that we would have `StrimziConnectContainer` (which would not be a part of the public API). It would be used to simplify the `StrimziConnectCluster` class, where you would do just the same as we do for StrimziKafkaCluster (i.e., start multiple StrimziKafkaContainer).